src/odemis/driver/autoscript_client.py

```diff
@@ -1770,12 +1792,31 @@ def terminate(self):
             self._pos_poll.cancel()
             self._pos_poll = None
 
+    def _get_coordinate_system_offset(self):
+        """Calculate the offset values for raw coordinate system. The offset is the difference between the specimen (linked) and raw coordinate system."""
+        self.parent.set_default_stage_coordinate_system("SPECIMEN")
+        pos_linked = self._getPosition()
+        self.parent.set_default_stage_coordinate_system("RAW")
+        pos = self._getPosition()
+        for axis in self._raw_offset.keys():
+            self._raw_offset[axis] = pos_linked[axis] - pos[axis]
+        # the offset should only be in linear axes, it is not expected in rotational axes
+        if not all(pos[axis] == pos_linked[axis] for axis in ["rx", "rz"]):
+            logging.warning(
+                "During offset estimation in linear axes in raw coordinate system in x and y, unexpected offset is "
+                "found in rotation axes")
```

Comment:
I have a hard time understanding what this warning means, can you write it down in a more clear way?

Reply:
better?